aten/src/ATen/test/vec256_test_all_types.cpp

```diff
@@ -0,0 +1,1098 @@
+
+        } //
+
+    }
+
+    template<typename vec_type, typename VT, int64_t mask>
+    typename std::enable_if_t<(mask < 0 || mask> 255), void>
+        test_blend(VT expected_val[vec_type::size()], VT a[vec_type::size()], VT b[vec_type::size()])
+    {
+    }
+
+    template<typename vec_type, typename VT, int64_t mask>
+    typename std::enable_if_t<(mask >= 0 && mask <= 255), void>
+        test_blend(VT expected_val[vec_type::size()], VT a[vec_type::size()], VT b[vec_type::size()])
+    {
+        //generate expected_val
+        int64_t m = mask;
+        for (int64_t i = 0; i < vec_type::size(); i++) {
+            if (m & 0x01) {
+                expected_val[i] = b[i];
+            }
+            else {
+                expected_val[i] = a[i];
+            }
+            m = m >> 1;
+        }
+        //test with blend
+        auto vec_a = vec_type::loadu(a);
+        auto vec_b = vec_type::loadu(b);
+        auto expected = vec_type::loadu(expected_val);
+        auto actual = vec_type::template blend<mask>(vec_a, vec_b);
+        std::function<std::string(int i)> detail = [](int i) {
+            std::stringstream stream;
+            stream << "mask " << mask << " index:" << i;
+            return stream.str();
+        };
+        AssertVec256(expected, actual, detail);
+        test_blend<vec_type, VT, mask - 1>(expected_val, a, b);
+    }
+
+
+    template<typename T, int N>
+    void blend_init(T(&a)[N], T(&b)[N]) {
+        a[0] = (T)1.0;
+        b[0] = a[0] + (T)N;
+        for (int i = 1; i < N; i++) {
+            a[i] = a[i - 1] + (T)(1.0);
+            b[i] = b[i - 1] + (T)(1.0);
+        }
+    }
+
+    template<>
+    void blend_init<Complex<float>, 4>(Complex<float>(&a)[4], Complex<float>(&b)[4]) {
+        auto add = Complex<float>(1., 100.);
+        a[0] = Complex<float>(1., 100.);
+        b[0] = Complex<float>(5., 1000.);
+        for (int i = 1; i < 4; i++) {
+            a[i] = a[i - 1] + add;
+            b[i] = b[i - 1] + add;
+        }
+    }
+
+    template<>
+    void blend_init<Complex<double>, 2>(Complex<double>(&a)[2], Complex<double>(&b)[2]) {
+        auto add = Complex<double>(1.0, 100.0);
+        a[0] = Complex<double>(1.0, 100.0);
+        b[0] = Complex<double>(3.0, 1000.0);
+        a[1] = a[0] + add;
+        b[1] = b[0] + add;
+    }
+
+    TYPED_TEST(BitwiseFloatsAdditional2, Blend) {
+        using vec_type = TypeParam;
+        using VT = ValueType<TypeParam>;
+        CACHE_ALIGN VT a[vec_type::size()];
+        CACHE_ALIGN VT b[vec_type::size()];
+        CACHE_ALIGN VT expected_val[vec_type::size()];
+        blend_init(a, b);
+        constexpr int64_t power_sets = 1 << (vec_type::size());
+        test_blend<vec_type, VT, power_sets - 1>(expected_val, a, b);
+    }
+
+
+
+
+    TEST(ComplexTests, TestComplexFloatImagRealConj) {
+        //vcomplex a = { Complex<float>(1,2),Complex<float>(3,4) ,Complex<float>(5,6) ,Complex<float>(6,7) };
```

Comment:
Is this a valid comment line? or should be deleted?

Reply:
will be removed